Gopkg.toml

```diff
@@ -22,16 +22,18 @@
 
 [[constraint]]
   name = "k8s.io/api"
-  version = "kubernetes-1.8.5"
-
-[[constraint]]
-  name = "k8s.io/apiextensions-apiserver"
-  version = "kubernetes-1.8.5"
+  version = "kubernetes-1.9.3"
 
 [[constraint]]
   name = "k8s.io/apimachinery"
-  version = "kubernetes-1.8.5"
+  version = "kubernetes-1.9.3"
 
 [[constraint]]
   name = "k8s.io/client-go"
-  version = "kubernetes-1.8.5"
+  version = "kubernetes-1.9.3"
+
+
+
+[[override]]
+  name = "k8s.io/apiextensions-apiserver"
+  version = "kubernetes-1.9.3"
```

Comment:
Why override? And why on the bottom of the file with 3 newlines? :)

Reply:
We have blocks like `requires`, `constrains`, `override`, `prune` and so on. I want to separate them visually a bit more.

Reply:
And O comes after C. You know. :D

Reply:
So far dep was apparently cool with a usual constraint. Now it complains the package is not used and if we want to ensure this specific version we have to set an override.

Reply:
AFAIK in that case `required` is your friend. I think `override` is helpful when you have conflicting dependencies.

https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md#override

Reply:
Argue with dep. It told me to define an override because the package defined here is not directly used. But we want to make sure the right dependency version is set. So I do what the tooling needs.

Reply:
OK I pulled the branch and made my tests. When I just change `override` to `constraint` it gives:

```
$ dep ensure
Warning: the following project(s) have [[constraint]] stanzas in Gopkg.toml:

  ✗  k8s.io/apiextensions-apiserver

However, these projects are not direct dependencies of the current project:
they are not imported in any .go files, nor are they in the 'required' list in
Gopkg.toml. Dep only applies [[constraint]] rules to direct dependencies, so
these rules will have no effect.

Either import/require packages from these projects so that they become direct
dependencies, or convert each [[constraint]] to an [[override]] to enforce rules
on these projects, if they happen to be transitive dependencies,
```

Which makes sense, because there isn't such import.

But with this content (added `required`) it works fine:

```
required = [
  "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
]

[[constraint]]
  name = "k8s.io/api"
  version = "kubernetes-1.9.3"

[[constraint]]
  name = "k8s.io/apiextensions-apiserver"
  version = "kubernetes-1.9.3"

[[constraint]]
  name = "k8s.io/apimachinery"
  version = "kubernetes-1.9.3"

[[constraint]]
  name = "k8s.io/client-go"
  version = "kubernetes-1.9.3"
```

After `dep ensure`:

```
$ dep status | grep k8s.io
k8s.io/api                           kubernetes-1.9.3  kubernetes-1.9.3  acf347b            28
k8s.io/apiextensions-apiserver       kubernetes-1.9.3  kubernetes-1.9.3  f142580            1
k8s.io/apimachinery                  kubernetes-1.9.3  kubernetes-1.9.3  19e3f5a            35
k8s.io/client-go                     kubernetes-1.9.3  kubernetes-1.9.1  9389c05            41
k8s.io/kube-openapi                  branch master     branch master     50ae88d   50ae88d  1
```

BTW I think `k8s.io/kube-openapi ` should be also `constraint`ed and `required`

Reply:
Note `dep`says `Either import/require` before `convert each [[constraint]] to an [[override]]`, and docs say this is last resort.

Reply:
So what is the problem with only having the override?

Reply:
I don't know, but when documentation says `Overrides should be used cautiously and temporarily, when possible.` I'm not keen to use them. I also think they especially shouldn't be used in libraries.

Reply:
I sorted this in the other PRs. This here goes into legacy which we want to get rid of anyway. Keep it like this now here. I do not mind. Give me a shout when this is wrong or important to you.